File: labs/2nd semestr/lab3/cs/string.cs
using System;

public class CustomString
{
    private string value;

    public CustomString()
    {
        value = "";
    }

    public CustomString(string str)
    {
        value = str;
    }

    public CustomString(CustomString other)
    {
        value = other.value;
    }

    public int Length()
    {
        return value.Length;
    }

    public string GetValue()
    {
        return value;
    }

    public static CustomString operator +(CustomString s1, CustomString s2)
    {
        return new CustomString(s1.value + s2.value);
    }

    public static CustomString operator +(CustomString s1, string s2)
    {
        return new CustomString(s1.value + s2);
    }

    public static CustomString operator -(CustomString s, char c)
    {
        return new CustomString(s.value.Replace(c.ToString(), ""));
    }

    public override string ToString()
    {
        return value;
    }
}

File: labs/1st course/2nd semestr/lab3/ts/rhombus.ts/string.ts
export class CustomString {
    private value: string;

    constructor(str: string = "") {
        this.value = str;
    }

    static copy(other: CustomString): CustomString {
        return new CustomString(other.value);
    }

    length(): number {
        return this.value.length;
    }

    getValue(): string {
        return this.value;
    }

    add(other: CustomString | string): CustomString {
        return new CustomString(this.value + (typeof other === "string" ? other : other.value));
    }

    subtract(charToRemove: string): CustomString {
        return new CustomString(this.value.split(charToRemove).join(""));
    }
}